core/src/window_service.rs

```diff
@@ -68,6 +92,44 @@ impl WindowServiceMetrics {
     }
 }
 
+#[derive(Default)]
+struct RecvWindowMetrics {
+    recv_window_count: u64,
+    num_packets_received: usize,
+    verified_receiver_recv_elapsed_us: u64,
+    handle_packets_elapsed_us: u64,
+    recv_window_elapsed_us: u64,
+}
+
+impl RecvWindowMetrics {
+    pub fn report_metrics(&self, metric_name: &'static str) {
+        datapoint_info!(
+            metric_name,
```

Comment:
Any value making this a parameter compared to a name like "recv_window_metrics"? The last_report logic can be implemented using the AtomicInterval? See ShrinkStats in accounts_db and other patterns.